pkg/apis/core/validation/validation.go

```diff
@@ -3024,10 +3022,70 @@ func validateInitContainers(containers []core.Container, otherContainers []core.
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("startupProbe"), ctr.StartupProbe, "must not be set for init containers"))
 		}
 	}
+
+	return allErrs
+}
+
+func validateContainer(ctr *core.Container, volumes map[string]core.VolumeSource, path *field.Path, opts PodValidationOptions) field.ErrorList {
```

Comment:
Since you are in here - some comments detailing how the container validation flow runs from pod->... thru these helper functions would be lovely? Even just a one-liner at each function explaining it..

Reply:
Done :+1: